### ISSUE TITLE
[BUG] harvester-node-manager-webhook chart incompatible with mcc-harvester bundle when number of nodes < desired replicas

### ISSUE THREAD

**Issue by connorkuehl** [harvester/harvester#5212]
**Describe the bug**

The harvester-node-manager-webhook chart was recently modified to use the `management.cattle.io/scale-available` annotation to automatically scale the deployment up to the desired number of replicas depending on the number of nodes in the cluster.

However, when the number of nodes in the cluster is less than the desired number of replicas, this causes the mcc-harvester bundle to become OutOfSync. This will block upgrades.

**To Reproduce**
1. Create a Harvester cluster with fewer nodes than the default number of harvester-node-manager-webhook replicas (which is 3)
2. Check the bundle status

```console
harv1:~ # kubectl -n fleet-local get bundle mcc-harvester
NAME            BUNDLEDEPLOYMENTS-READY   STATUS
mcc-harvester   0/1                       Modified(1) [Cluster fleet-local/local]; deployment.apps harvester-system/harvester-node-manager-webhook modified {"spec":{"replicas":3}}
```

**Expected behavior**

The mcc-harvester bundle should be shown as Ready.

**Additional context**

Introduced by: https://github.com/harvester/harvester/issues/5149

Originally reported: https://github.com/harvester/harvester/issues/5149#issuecomment-1960852535

---

- [x] Backport fix to release/ branch
- [x] Update vendored node-manager chart in harvester/harvester

**Comment by w13915984028** [comment]
@connorkuehl Could the node-manager-webhook just deploy 1 replica instead of the default 3? most of the configuration changes in Harvesters are not node-manager related.

**Comment by connorkuehl** [comment]
> Could the node-manager-webhook just deploy 1 replica instead of the default 3? most of the configuration changes in Harvesters are not node-manager related.

@w13915984028 I don't see any real issues with that but I have no strong preference either way.

**Comment by connorkuehl** [comment]
Fix should be available as of 1.3 commit d60468f14c49cd5c45c9adfa3081a112c0a4af51.

Moving to Ready for Testing.

**Comment by albinsun** [comment]
Provide test result on `v1.3-d60468f1-head`, close as fixed.

## Environment
* Harvester
  * Version: `v1.3-d60468f1-head`
  * Profile: QEMU/KVM, 
  * ui-source: `Auto`


## Steps 
1. Setup default node-0
   <details>
   <summary> <b>mcc-harvester</b> bundle should be <code>Ready</code> :heavy_check_mark: 
   </summary>

   ![image](https://github.com/harvester/harvester/assets/2773781/54936ea6-18df-490a-8d86-501f5b1bc299)   
   </details>


1. Join default node-1
   <details>
   <summary> <b>mcc-harvester</b> bundle should be <code>Ready</code> :heavy_check_mark:
   </summary> 

   ![image](https://github.com/harvester/harvester/assets/2773781/975ab951-e6f8-462f-a75b-e889bbeba602) 
   </details>

   

1. Join witness node-2
   <details>
   <summary> <b>mcc-harvester</b> bundle should be <code>Ready</code> :heavy_check_mark:
   </summary> 

   ![image](https://github.com/harvester/harvester/assets/2773781/1cd1ac0f-be22-4e35-b1bb-347bef9a19af)
   </details>


1. Join worker node-3
   <details>
   <summary> <b>mcc-harvester</b> bundle should be <code>Ready</code> :heavy_check_mark:
   </summary> 

   ![image](https://github.com/harvester/harvester/assets/2773781/52fa87c2-4b49-4305-8a96-ad2324154ab2)   
   </details>


1. Delete witness node-2
   <details>
   <summary> <b>mcc-harvester</b> bundle should be <code>Ready</code> :heavy_check_mark:
   </summary> 

   ![image](https://github.com/harvester/harvester/assets/2773781/f78a833a-3b96-40fb-81e9-e1ab53fdc6dd)  
   </details>

1. Join default node-4
   <details>
   <summary> <b>mcc-harvester</b> bundle should be <code>Ready</code> :heavy_check_mark:
   </summary> 

   ![image](https://github.com/harvester/harvester/assets/2773781/7340fb99-39a2-413a-b36b-16ef16b754b7)  
   </details>

1. Delete default node-1
   <details>
   <summary> <b>mcc-harvester</b> bundle should be <code>Ready</code> :heavy_check_mark:
   </summary> 

   ![image](https://github.com/harvester/harvester/assets/2773781/3cee4597-a508-44b1-9dff-faa8f3994b1d)
   </details>

1. Delete default node-4
   <details>
   <summary> <b>mcc-harvester</b> bundle should be <code>Ready</code> :heavy_check_mark:
   </summary> 

   ![image](https://github.com/harvester/harvester/assets/2773781/b143c5ba-90b7-4233-8053-dc0a14ca4498)
   </details>


1. Delete worker node-3
   <details>
   <summary> <b>mcc-harvester</b> bundle should be <code>Ready</code> :heavy_check_mark:
   </summary> 

   ![image](https://github.com/harvester/harvester/assets/2773781/eaa622f0-e5ce-4ff8-91d5-5857db825bb1) 
   </details>
